openedx/core/djangoapps/user_authn/views/register.py

```diff
@@ -519,18 +538,46 @@ def _create_account(self, request, data):
                 field: [{"user_message": error} for error in error_list]
                 for field, error_list in err.message_dict.items()
             }
-            response = self._create_response(errors, status_code=400)
+            response = self._create_response(request, errors, status_code=400)
         except PermissionDenied:
             response = HttpResponseForbidden(_("Account creation not allowed."))
 
         return response, user
 
-    def _create_response(self, response_dict, status_code):
+    def _create_response(self, request, response_dict, status_code):
         if status_code == 200:
             # keeping this `success` field in for now, as we have outstanding clients expecting this
             response_dict['success'] = True
+        else:
+            self._log_validation_errors(request, response_dict, status_code)
+
         return JsonResponse(response_dict, status=status_code)
 
+    def _log_validation_errors(self, request, errors, status_code):
+        if not REGISTRATION_FAILURE_LOGGING_FLAG.is_enabled():
+            return
+
+        try:
+            user_agent = request.META.get('HTTP_USER_AGENT', '').lower()
+            if 'iphone' in user_agent:
```

Comment:
Not really sure if its by-design or a bug but recently I haven't been able to see the `iPhone` string in requests made by our iPhone app on Splunk. Another strange thing is `user-agent` field not appearing in our Splunk logs for requests coming from iPhone app.

So this imo will result in `agent_category` set to `ios` for requests coming from a browser on an iPhone.

Reply:
Interesting. I've removed the categorization so that we can do that kind of thing in splunk at analysis time.

Reply:
Instead I opted to just include the whole user agent